.github/workflows/extra-rule-validation.yml

```diff
@@ -0,0 +1,58 @@
+name: ⚙️ Extra Rule Validation
```

Comment:
As we've renamed this workflow, do we need to update the required checks for branch protection?

Reply:
> Looks good to me, I think we just need to fix the existing instances.
> 
> Note: in the DeadCode package I added some standard text to the top of shared files: https://github.com/github/codeql-coding-standards/pull/163/files#diff-dffa7a1661d2b5c024e7778c5880d0fc9915002462aa5f9cae5dba120692361aR1-R3 Noting that test cases may be duplicated between the C and C++ tests.

I saw, thank you for doing that. The text will be super helpful if all cases had it!

Reply:
Also, not sure about what the branch protections will do -- I'll check it out once this merges.